flang/test/Lower/percent-val-actual-argument.f90

```diff
@@ -0,0 +1,14 @@
+! RUN: bbc %s -emit-fir -o - | FileCheck %s
+
+program main
+  logical::a1
+  data a1/.true./
+  call sa(%val(a1))
+! CHECK: fir.load %3 : !fir.ref<!fir.logical<4>>
+! CHECK: fir.convert %13 : (!fir.logical<4>) -> i1
+  write(6,*) "a1 = ", a1
+end program main
+
+subroutine sa(x1)
+  logical::x1
+end subroutine sa
```

Comment:
Please could you add a second test to make sure that we continue to correctly handle passing %VAL to a VALUE argument. Something like

```
program main
  ...
  call sb(%val(a1))
  ...
contains
  subroutine sb(x1)
    logical, value :: x1
  end subroutine sb
end program main
```